src/execution/reservoir_sample.cpp

```diff
@@ -49,56 +64,193 @@ void ReservoirSample::AddToReservoir(DataChunk &input) {
 	}
 }
 
-unique_ptr<DataChunk> ReservoirSample::GetChunk() {
-	if (!reservoir_chunk || reservoir_chunk->size() == 0) {
+void ReservoirSample::Merge(unique_ptr<BlockingSample> other) {
+	D_ASSERT(other->type == SampleType::RESERVOIR_SAMPLE);
+	auto reservoir_other = &other->Cast<ReservoirSample>();
+	// three ways to merge samples
+	if (sample_count != reservoir_other->sample_count) {
+		throw NotImplementedException("attempting to merge samples with different sample counts");
+	}
+
+	// There are four combinations for reservoir state
```

Comment:
There are four combinations for reservoir state when merging to reservoir samples.